### ISSUE TITLE
Bug: Content parsing failure with serial matches

### ISSUE THREAD

**Issue by shawnkhall** [dnGrep/dnGrep#601]
Using dnGrep 2.9.428.0

There is a bug in the search results that causes the preview text to display corrupted information when the search term appears two or more times in succession, or overlapping within first match. This is easiest to recreate when you search for 2 or more of the same character and the matching line has 3 or more instances of the same character in a row.

`Highlight Matches` does not need to be enabled for this to occur.

I haven't yet pinned down exactly what the cause is, but when performing a `Text` search within files for `99` and the text `999` appears in the file, the entire contents of the line is duplicated immediately after the first match point within the search results for that line, and again for each additional match on the line (ad infinitum). 

![dngrep-search-99](https://user-images.githubusercontent.com/6037167/147191008-3c1a4b48-058b-4f4a-a3cd-b41201b3dec0.png)

In the actual file, line 38 reads:
`:: 100-999`

Line 39 reads:
`FOR /L %%s in (100,1,999) DO (`

This duplication occurs in at least a dozen different files in this search result. This sample is just the simplest to demonstrate the problem.

This bug appears to exist whenever a two-character search term is the same two characters and the results include 3 or more of that character in a row. I've tested with `==`, `AA` and `77` with the same results as the image.

It occurs with other instances of duplication as well. For example, when searching for "0.0" and the text "0.0.0.0" appears in a file it will duplicate the line at the match point twice.

To recreate the problem you can create a simple text file with the following contents:
`0.0.0.0`
Now search `Text` for `0.0` with the path to the file.  The results will be:
![dngrep-search-0dot0](https://user-images.githubusercontent.com/6037167/147193158-f6d5dbdf-c788-4d03-baa4-73c59785bb22.png)

**Comment by doug24** [comment]
Well, that's strange.  It will be interesting to see what happened to the code in there.

**Comment by doug24** [comment]
Fixed in 2.9.482

I was sure I had tested this scenario before, but looking back in the history, I don't see where it was working.

There were two bugs: first, highlighting the overlapping matches was failing and throwing an exception.  Second, the exception handler was supposed to show the whole line with no highlights, but instead was appending the whole line to end of the what had already been processed.
